### ISSUE TITLE
postgresql_server: change sku_name tier: error: InvalidUpgradeEdition

### ISSUE THREAD

**Issue by thomas-riccardi** [terraform-providers/terraform-provider-azurerm#7131]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.25
+ provider.azurerm v2.9.0
```

### Affected Resource(s)

* `azurerm_postgresql_server`

### Terraform Configuration Files

```
Terraform will perform the following actions:

  # azurerm_postgresql_server.vesta will be updated in-place
  ~ resource "azurerm_postgresql_server" "xxx" {
        administrator_login               = "xxx"
        administrator_login_password      = (sensitive value)
        auto_grow_enabled                 = true
        backup_retention_days             = 7
        create_mode                       = "Default"
        fqdn                              = "xxx.postgres.database.azure.com"
        geo_redundant_backup_enabled      = false
        id                                = "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.DBforPostgreSQL/servers/xxx"
        infrastructure_encryption_enabled = false
        location                          = "westeurope"
        name                              = "xxx"
        public_network_access_enabled     = true
        resource_group_name               = "xxx"
      ~ sku_name                          = "B_Gen5_1" -> "GP_Gen5_4"
        ssl_enforcement                   = "Enabled"
        ssl_enforcement_enabled           = true
        ssl_minimal_tls_version_enforced  = "TLSEnforcementDisabled"
        storage_mb                        = 512000
        tags                              = {}
        version                           = "11"

        storage_profile {
            auto_grow             = "Enabled"
            backup_retention_days = 7
            geo_redundant_backup  = "Disabled"
            storage_mb            = 512000
        }
    }
```

### Expected Behavior

Successfully update the SKU.

It doesn't seem to be possible to update the `tier` (aka `edition`) (indeed in the portal it's not doable), so probably the provider should plan to delete and re-create the database server resource.

### Actual Behavior
```
Error: updating PostgreSQL Server "foo" (Resource Group "foo"): postgresql.ServersClient#Update: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidUpgradeEdition" Message="The edition 'GeneralPurpose' is not a valid edition. Edition cannot be changed by update."

  on resources.tf line 21, in resource "azurerm_postgresql_server" "foo":
  21: resource "azurerm_postgresql_server" "foo" {
```

### Steps to Reproduce

1. `terraform apply` with sku `B_*`
2. `terraform apply` with sku `GP_*`

**Comment by tombuildsstuff** [comment]
Fixed via #7456

**Comment by thomas-riccardi** [comment]
@tombuildsstuff thanks! 

According to https://techcommunity.microsoft.com/t5/azure-database-for-mysql/upgrade-from-basic-to-general-purpose-or-memory-optimized-tiers/ba-p/830404 there seems to be the same issue for mysql servers, and possibly for other services too (I don't know azure well enough). It may be a good idea to try to fix them all.
